src/EnergyPlus/HeatBalanceSurfaceManager.cc

```diff
@@ -2544,17 +2546,17 @@ void InitSolarHeatGains(EnergyPlusData &state)
             state.dataSurface->SurfSkySolarInc(SurfNum) = 0.0;
             state.dataSurface->SurfGndSolarInc(SurfNum) = 0.0;
         }
-        for (int zoneNum = 1; zoneNum <= state.dataGlobal->NumOfZones; ++zoneNum) {
-            state.dataHeatBal->ZoneTransSolar(zoneNum) = 0.0;
-            state.dataHeatBal->ZoneBmSolFrExtWinsRep(zoneNum) = 0.0;
-            state.dataHeatBal->ZoneBmSolFrIntWinsRep(zoneNum) = 0.0;
-            state.dataHeatBal->ZoneDifSolFrExtWinsRep(zoneNum) = 0.0;
-            state.dataHeatBal->ZoneDifSolFrIntWinsRep(zoneNum) = 0.0;
-            state.dataHeatBal->ZoneTransSolarEnergy(zoneNum) = 0.0;
-            state.dataHeatBal->ZoneBmSolFrExtWinsRepEnergy(zoneNum) = 0.0;
-            state.dataHeatBal->ZoneBmSolFrIntWinsRepEnergy(zoneNum) = 0.0;
-            state.dataHeatBal->ZoneDifSolFrExtWinsRepEnergy(zoneNum) = 0.0;
-            state.dataHeatBal->ZoneDifSolFrIntWinsRepEnergy(zoneNum) = 0.0;
+        for (int enclNum = 1; enclNum <= state.dataViewFactor->NumOfSolarEnclosures; ++enclNum) {
```

Comment:
Stumbled across places in the code that should be dimensioned by enclosure rather than zone and vice versa.